cpp/src/Slice/Parser.h

```diff
@@ -397,6 +397,18 @@ namespace Slice
 
         static FormatType parseFormatMetaData(const std::list<std::string>&);
 
+        /// Returns true if this item is deprecated (due to the presence of `deprecate` metadata).
+        /// @param checkParent If true, this item's immediate container will also be checked for `deprecate` metadata.
+        /// @return True if this item (or possibly its container) has `deprecate` metadata on it, false otherwise.
+        bool isDeprecated(bool checkParent) const;
+
+        /// If this item is deprecated, return its deprecation message (if present).
```

Comment:
I don't like the name of this function.
`getDeprecatedReason` or `getDeprecationMessage` would be better.

Reply:
Sure, I'll switch it to `getDeprecationMessage`

Reply:
Fixed.